File: JanperCab/5-JanperCab/ClientApp/src/app/_models/DuraformProcess/DuraformProcessRoutingDto.ts
import { MachineRouterDto } from './../machine/MachineRouterDto';
import { DuraformProcessDto } from './DuraformProcessDto';

export class DuraformProcessRoutingDto extends DuraformProcessDto {
  machineId: number;

  machineRouter: MachineRouterDto;

  getStatus(): string {
    status = this.startTime && this.endTime ? 'Routed' : 'Routing';

    if (this.machineRouter) {
      status += ` - ${this.machineRouter.name}`;
    }

    return status;
  }
}

File: JanperCab/3-Application/Dtos/Process/DuraformProcessRoutingDto.cs
using _3_Application.Dtos.Machine;

namespace _3_Application.Dtos.Process
{
    public class DuraformProcessRoutingDto : DuraformProcessDto
    {
        public int? MachineId { get; set; }

        public MachineDto MachineRouter { set; get; }
    }
}